client/src/unifycr-sysio.c

```diff
@@ -1038,16 +1044,25 @@ int UNIFYCR_WRAP(lio_listio)(int mode, struct aiocb *const aiocb_list[],
     for (i = 0; i < nitems; i++) {
         if (aiocb_list[i]->aio_lio_opcode != LIO_READ) {
             //does not support write operation currently
+            free(reqs);
+            errno = EIO;
             return -1;
         }
-        reqs[i].fid    = aiocb_list[i]->aio_fildes;
-        reqs[i].buf    = (char *)aiocb_list[i]->aio_buf;
-        reqs[i].length = aiocb_list[i]->aio_nbytes;
-        reqs[i].offset = aiocb_list[i]->aio_offset;
+        reqs[i].fid     = aiocb_list[i]->aio_fildes;
+        reqs[i].offset  = aiocb_list[i]->aio_offset;
+        reqs[i].length  = aiocb_list[i]->aio_nbytes;
+        reqs[i].errcode = UNIFYCR_SUCCESS;
+        reqs[i].buf     = (char *)aiocb_list[i]->aio_buf;
     }
 
     int ret = unifycr_fd_logreadlist(reqs, nitems);
 
+    for (i = 0; i < nitems; i++) {
+        /* TODO: update fields to record error status
+         * see /usr/include/aio.h,
+         * update __error_code and __return_val fields? */
+    }
+
```

Comment:
@adammoody I'm confused about what this loop does exactly? Looks like it isn't functional right now, but would this be updating the errcode field you added to the structs?

Reply:
Long story on that non-functioning loop... my intent was to set the error codes in each of the aiocb structs based on the errors in the corresponding read requests that were passed to and modified by fd_logreadlist.  However, I realized it was not straight-forward.  With aio calls, one should call aio_error() and then aio_return() to finally get the return code.  The easiest thing would be to save the error code on a field in the aiocb struct, but it's not clear which fields on the aiocb struct are portable for that use.  Because it was going to take a lot more thought, I decided to defer that to a later PR.  I left this loop in there as a reminder as one of the things that needs to be looked at.  I thought I created an issue, too, but I'm not seeing it.  I'll do that if I never did.

Reply:
Created an issue to address aio wrappers in general, which will include this fix: https://github.com/LLNL/UnifyCR/issues/225